roles/network_plugin/calico/tasks/install.yml

```diff
@@ -347,3 +348,30 @@
   when:
     - peer_with_router|default(false)
     - inventory_hostname in groups['k8s-cluster']
+    - item.nodeSelector is not defined
+
+- name: Calico | Configure peering with router(s) via nodeSelectors
```

Comment:
@catblade I think you might take a look at how the global peers are defined, adding the same condition so that the task is run only on the first master would do the trick.
And maybe add a `selectattr('nodeSelector','defined')` filter in the loop